### ISSUE TITLE
[BUG] Trying to animate the z-index of element using keyframes.

### ISSUE THREAD

**Issue by mrbjjackson** [framer/motion#1254]
**Describe the bug**

I'm trying to create a looping animation where one element "orbits" around the other one. This requires me to change the z-index of the elements as they move across one another.

I've tried to use an array of keyframes but this is producing strange results (i.e. if I use `zIndex: [5, 15]` then the z-index property of the element does not animate - it just is set to `z-index="5, 15"`)

**IMPORTANT: Provide a CodeSandbox reproduction of the bug**

[Sandbox example can be found here](https://codesandbox.io/s/framer-motion-z-index-issue-rsj8l?file=/src/App.js)

**Steps to reproduce**

The issue should be obvious from the Sandbox - after the "moon" passes _in front_ of the "earth" and passes back, it should be _behind_ the earth

**Expected behavior**

I realise that things probably get tricky with z-index because it works in discreet integers rather than decimals but I was expecting that framer-motion had a way of dealing with it (i.e. by rounding to the nearest integer)

Thank you

**Comment by mrbjjackson** [comment]
Thanks very much in advance for any help you can offer on this issue.

**Comment by mattgperry** [comment]
`zIndex` is explicitly set as non-animatable as people quite often use it in variants and we had a load of reports about this not being the desired behaviour. For your use-case you should be able to set `zIndex` as a motion value and animate is explicitly with `animate`

```
import { animate, useMotionValue, motion } from "framer-motion"

function Component() {
  const zIndex = useMotionValue(0)

  useEffect(() => {
    animate(5, 15, { duration: 2, onUpdate: latest => zIndex.set(Math.round(latest)) })
  })
  
  return (
    <motion.div style={{ zIndex }} />
  )
}
```

**Comment by mrbjjackson** [comment]
Thanks so much Matt, you are a legend. 

FYI - I think I found another way around it too:

By using `"z-index": [5, 15]` (instead of zIndex) I was able to get a z-index attribute to animate as desired on the svg element.

Weirdly though this wasn't making any difference to the appearance of the animation. However, after doing some debugging it appears that adding "z-index" as an _attribute_ on an svg element doesn't do anything, it needs to be in the style attribute. However I wrapped the svg in a div and animated the z-index on that and it looks like it's working.

Thanks again.
